app/controllers/smart_proxies_controller.rb

```diff
@@ -159,4 +161,19 @@ def resource_base
     base = base.eager_load(:organizations) if SETTINGS[:organizations_enabled]
     base
   end
+
+  def versions_mismatched?(proxy_versions)
+    foreman_version = Foreman::Version.new.notag
+    proxy_version = Foreman::Version.new(proxy_versions['version']).notag
+
+    foreman_version != proxy_version
+  end
+
+  def version_mismatch_warning(versions_hash)
```

Comment:
Maybe call both this and the parameter in `versions_mismatched` proxy_versions_hash?
I know this is a huge nitpick but I did not understand why you did `['version']` twice, hash in hash is confusing :)